third_party/blake3/BUILD

```diff
@@ -0,0 +1,83 @@
+load("@rules_license//rules:license.bzl", "license")
+load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library")
+
+licenses(["notice"])  #  BSD/MIT-like license
+
+exports_files(["LICENSE"])
+
+license(
+    name = "license",
+    package_name = "blake3",
+    license_kinds = [
+        "@rules_license//licenses/spdx:Apache-2.0",
+    ],
+    license_text = "LICENSE",
+    package_version = "1.3.3",
+)
+
+filegroup(
+    name = "srcs",
+    srcs = glob(["**"]),
+    visibility = ["//third_party:__pkg__"],
+)
+
+cc_library(
+    name = "blake3",
+    srcs = [
+        "src/c/blake3.c",
+        "src/c/blake3_dispatch.c",
+        "src/c/blake3_portable.c",
+    ] + select({
+        "//src/conditions:linux_x86_64": [
+            "src/c/blake3_avx2_x86-64_unix.S",
+            "src/c/blake3_avx512_x86-64_unix.S",
+            "src/c/blake3_sse2_x86-64_unix.S",
+            "src/c/blake3_sse41_x86-64_unix.S",
+        ],
+        "//src/conditions:windows_x86_64": [
+            "src/c/blake3_avx2_x86-64_windows_msvc.asm",
+            "src/c/blake3_avx512_x86-64_windows_msvc.asm",
+            "src/c/blake3_sse2_x86-64_windows_msvc.asm",
+            "src/c/blake3_sse41_x86-64_windows_msvc.asm",
+        ],
+        "//src/conditions:darwin_arm64": [
+            "src/c/blake3_neon.c",
+        ],
+        "//conditions:default": [],
```

Comment:
I'm not familiar with BLAKE3's source code. Are these assembly code optional so that when targeting other platforms the code can still be compiled but is just slower? I guess the `copts` below answers my question but just want to double check.

cc @meteorcloudy for potential caring when targeting other platforms.

Reply:
Yes, you got it. The blake3_portable.c implementation is a safe fallback for architectures which do not support the vectorized ASM implementations.